src/commands/init.ts

```diff
@@ -0,0 +1,167 @@
+  },
+} satisfies Record<string, FlaggablePrompt>
+
+export default class Generate extends GeneratorCommand<typeof Generate> {
+  static description = `This will clone the template repo and update package properties. For CommonJS, the 'oclif/hello-world' template will be used and for ESM, the 'oclif/hello-world-esm' template will be used.`
+
+  static examples = [
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli',
+      description: 'Generate a new CLI with prompts for all properties',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --yes',
+      description: 'Automatically accept default values for all prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe"',
+      description: 'Supply answers for specific prompts',
+    },
+    {
+      command: '<%= config.bin %> <%= command.id %> my-cli --module-type CommonJS --author "John Doe" --yes',
+      description: 'Supply answers for specific prompts and accept default values for the rest',
+    },
+  ]
+
+  static flaggablePrompts = FLAGGABLE_PROMPTS
+
+  static flags = {
+    ...makeFlags(FLAGGABLE_PROMPTS),
+    'module-type': Flags.option({
+      options: ['CommonJS', 'ESM'],
+    })({}),
+  }
+
+  static summary = 'Initialize a new oclif CLI'
+
+  async run(): Promise<void> {
+    const outputDir = await this.getFlagOrPrompt({defaultValue: './', name: 'output-dir', type: 'input'})
+    const location = resolve(outputDir)
+
+    this.log(`Initializing oclif in ${chalk.green(location)}`)
+
+    const packageJSON = (await readPJSON(location))!
+    if (!packageJSON) {
+      throw new Errors.CLIError(`Could not find a package.json file in ${location}`)
+    }
+
+    const bin = await this.getFlagOrPrompt({
+      defaultValue: location.split(sep).at(-1)!,
+      name: 'bin',
+      type: 'input',
+    })
+
+    let moduleType = this.flags['module-type']
+    if (!moduleType) {
+      if (packageJSON.type === 'module') {
+        moduleType = 'ESM'
+      } else if (packageJSON.type === 'commonjs') {
+        moduleType = 'CommonJS'
+      } else {
+        moduleType = await (
+          await import('@inquirer/select')
+        ).default({
+          choices: [
+            {name: 'CommonJS', value: 'CommonJS'},
+            {name: 'ESM', value: 'ESM'},
+          ],
+          message: 'Select a module type:',
+        })
+      }
+    }
+
+    this.log(`Using module type ${chalk.green(moduleType)}`)
+
+    const template = moduleType === 'ESM' ? 'hello-world-esm' : 'hello-world'
+    const repoPath = join(location, '_tmp-template')
+    await clone(template, repoPath)
+
+    const repoBinPath = join(repoPath, 'bin')
+    const binFiles = await readdir(repoBinPath)
+    for (const binFile of binFiles) {
+      const filePath = join(repoBinPath, binFile)
+      if (statSync(filePath).isFile()) {
```

Comment:
Using sync here because async fails aslant checks. Options here are:

- Keep this as-is
- Use `async` and ignore or change eslint config
- Explicitly move the 4 files